src/main/java/it/gov/pagopa/payhub/activities/service/ErrorArchiverService.java

```diff
@@ -0,0 +1,126 @@
+package it.gov.pagopa.payhub.activities.service;
+
+import it.gov.pagopa.payhub.activities.dto.ingestion.IngestionFlowFileErrorDTO;
+import it.gov.pagopa.payhub.activities.exception.NotRetryableActivityException;
+import it.gov.pagopa.payhub.activities.util.Utilities;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.CollectionUtils;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * A base service for archiving error files.
+ * This class provides common functionality for writing error data to CSV files and archiving them into ZIP archives.
+ * It is designed to be extended by specific error archiver services for different types of error DTOs.
+ *
+ * @param <T> The type of the error DTO.
+ */
+@Slf4j
+public abstract class ErrorArchiverService<T extends IngestionFlowFileErrorDTO> {
```

Comment:
Let's move and rename the IngestionFlowFileErrorDTO with a more generic name, like `ErrorFileDTO`